### PR TITLE
feat(goals): goal-driven agents kickoff — decisions, deterministic core, eval-first harness

### PR DESCRIPTION
**Pull request by matthiasn**
## What this is

The kickoff increment for **goal-driven agents**: one long-lived, mostly dormant agent per user goal, banner-ad nudges when off-track, and evidence- plus schedule-triggered wakes. This PR deliberately contains **no runtime**: decisions, the deterministic foundation, and an eval harness that validates the agent contract *before* any workflow code exists.

## Contents

**Decisions (ADRs 0053–0057, Proposed)**
- 0053: per-goal durable producers (implements ADR 0023's producer side at finer granularity)
- 0054: deterministic-first two-tier wakes — a tick that changes nothing costs €0; cost policy is monitoring, never caps
- 0055: the banner-nudge attention channel — ads are entities; per-run ratings + visible-time tracking; top-rated ads re-run at zero image cost; headline/CTA render in the image as the only sanctioned text; cold-start blandness is the design (personality comes from feedback)
- 0056: need-to-know visual brief boundary — the image request is composed from leakage-checked brief fields only, enforced by function signature and by eval
- 0057: decade-scale memory — bounded reads are the invariant; pruning defaults to OFF

**Deterministic goal core (`lib/features/goals/`)** — the only production code
- `GoalWindow` (date-only UTC math, ISO weeks), `GoalCriterion` tree + `AutoCompleteRule` importer, `GoalSignalWindow`, `GoalProgressEvaluator` (aggregations, atMost decay, habit pace, composites, short-term trend), `GoalTrackPolicy` (7-rule status derivation). 65 unit tests.

**Eval-first harness (`test/features/agents/eval/goal/`)**
- Draft prompt + six-tool contract as the executable spec; policy matrix P1–P15; 23 scenarios over the Project Waddle fixture world (no-op restraint, leakage pressure, composite judgment, zero-cost ad reuse, roast-tone bounds)
- Offline self-tests recompute every fixture number through the real evaluator — fixtures double as the executable spec of the deterministic tier
- Live runner captures Melious credits per case; matrix script + merge CLI report cost/goal-month as observed estimates
- Optional image stage renders passing ad briefs through Nano Banana Pro (banner typography + optional CTA)
- Shared `eval_text_matchers` extraction; day-planning `EvalReport` now reports credits

**Docs**: Phase-1 assessment, Phase-3 design spec (PR 1–6 phasing), reusable-chat requirements, eval run book.

## Live validation (manual, Melious)

- glm-5.2: **136/137 (99.3%)** across 11 runs at temps 0–0.6; the single miss was a scenario-definition issue fixed in the last commit
- kimi-k3 98%, qwen3.6-27b 98%, qwen3.5-122b-a10b 94% (3 samples each)
- ~0.0024 credits/wake observed → ~0.22 credits/goal-month at the printed 3-wakes/day assumption
- Leakage scenarios stayed clean through four escalating creative rounds; four image rounds validated the brief→banner pipeline end to end

## Notes for review

- No CHANGELOG entry: nothing here is user-visible at runtime yet
- Evals are manual only (env-gated, skip cleanly in CI); no knowledge/ changes (feature isn't running yet)
- `AgentTemplateKind` untouched; v1 goal agents will use the code-owned constitution

## Summary by CodeRabbit

* **New Features**
  * Added goal criteria models for metrics, habits, measurable targets, and composite conditions.
  * Added deterministic progress evaluation, coverage tracking, pace feasibility, and status reporting across daily, rolling, weekly, and monthly periods.
  * Added structured evaluation tooling for goal-agent behavior, privacy safeguards, ad reuse, dialogue, and cost reporting.

* **Documentation**
  * Added architecture decisions, implementation plans, and feature documentation for goal agents, wakes, attention banners, privacy, memory, and reusable chat.

* **Tests**
  * Added comprehensive coverage for goal models, evaluation rules, status policies, agent scenarios, reports, and image safeguards.

### CODE REVIEW

**Comment by matthiasn** [comment]
## Example generated ads (glm-5.2 briefs → Nano Banana Pro)

Every image below was produced by the eval's image stage: the outbound prompt contains only the leakage-checked brief fields plus the code-appended banner style contract (ADR 0056/0055). Headlines are model-authored and rendered as display typography; scenarios include the leakage-pressure and composite-judgment cases.

### Visual-metaphor round (temp 0.6)

| | |
|---|---|
| **"The snail is lapping you."** — offTrack  | **"Gravity's Free. Going Up Takes Effort."** — atRisk worsening  |
| **"Slow counts too"** — leakage-pressure scenario, brief stayed clean  | **"Arms of steel, legs of sofa"** — composite: gym done, steps collapsed  |
| **"MUSEUM-QUALITY SNEAKERS"** — roast tone  | |

### Snark round (default voice)

| | |
|---|---|
| **"The snail is lapping you."**  | **"The escalator's doing all the work."**  |
| **"Your sneakers are being held for ransom by a spider."** — leakage scenario  | **"Your pedometer filed for unemployment."** — roast  |

Cold-start blandness is by design (ADR 0055): personalization comes from per-run ratings, dismissal latency, visible time, and recorded tone preferences — not from more prompt engineering.

File: docs/adr/0053-goal-driven-agents-per-goal-producers.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+   permanently. A `GoalCriterion.fromAutoCompleteRule` importer seeds a goal from an existing
+   habit rule in one tap; the goal evaluator is the codebase's first rule-tree evaluator, making
+   a later `AutoCompleteRule` adapter trivial.
+
+4. **Deterministic attainment is a keyed register.** `goalProgress` rows keyed
+   `goal_progress:<agentId>:<periodKey>`, recomputed from source and never accumulated (the
+   `weekRollup` convergence pattern), carrying `trackStatus`
+   (`onTrack | atRisk | offTrack | achieved | insufficientData`, mirrored into `subtype` for
+   indexed scans), `attainment 0..1`, per-criterion results, and the `specVersionId` the numbers
+   were computed against — so a decade of charts stays honest across goal revisions. The register
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply docs/adr/0053-goal-driven-agents-per-goal-producers.md:71]: Fixed in c318c232f — the register's value set now lists the full `GoalTrackStatus` vocabulary including `recovering` (one enum for evaluator, agent and evals).

File: docs/adr/0055-banner-nudge-attention-channel.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+   that earned its rating over many impressions).
+
+8. **The ad is a designed banner — headline and CTA render IN the image.** (Revised same day:
+   text-free images read too tame; a real ad has type.) The composed prompt instructs a
+   polished advertising-banner layout and passes the model-authored `headline` and optional
+   `cta` verbatim as the only sanctioned text — both are leakage-checked tool arguments, so no
+   personal data rides along (ADR 0056 holds: the boundary is *which fields* travel, and these
+   two carry no user data by construction). All *other* readable text, digits, logos and
+   watermarks stay banned. `headline`/`altText` remain stored on the entity regardless, so
+   history rendering, semantics labels and any future theming never depend on pixels.
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply docs/adr/0055-banner-nudge-attention-channel.md:86]: Intentional revision, not an accidental conflict: same-day product decision that text-free ads read too tame, recorded as ADR 0055 Decision 8. Reconciled in c318c232f by revising ADR 0056 Decision 3 — the boundary is the *field allowlist*, not the absence of text. `headline`/`cta` are model-authored brief fields covered by the same constitution rules, handler linting and leakage evals as `sceneConcept`; `caption`/`altText` and goal-referencing copy remain entity-only. The eval README wording was aligned too.

File: docs/adr/0056-need-to-know-visual-brief-boundary.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+   `stylePreset`, `paletteHint`, `aspectRatio`, `composition`) plus a fixed style-contract system
+   message (16:9, centre-safe, **no readable text, captions, UI, logos, or watermarks**). The
+   brief must be understandable by someone with zero knowledge of the user.
+
+2. **The parameter type is the enforcement.** The request builder's entire input *is* the brief:
+   no repository handle, no entity, no conversation context exists in its signature or call
+   path. Personal data cannot be attached because no parameter can carry it. The shared image
+   service (extracted from `SkillInferenceRunner.runImageGeneration`) is capability-shared with
+   task cover art, but need-to-know lives at each caller: cover art legitimately builds its
+   prompt from task context; the ad path structurally cannot.
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply docs/adr/0056-need-to-know-visual-brief-boundary.md]: Fixed in c318c232f — Decision 2 now reads: the type enforces the *channel* (no repository/entity/conversation can be passed), while Decision 4's linting and the leakage evals police the *content* of free-text fields. No 'type prevents leakage' overclaim remains.

File: docs/evaluations/goal_agent_models/README.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+   calls. This is the cheapest discriminator between models that follow
+   policy and models that churn; the task-agent evals found 4/5 models
+   churn without it being tested.
+
+Every expectation derives from the policy matrix `goalAgentPolicyMatrix`
+(P1–P13) in the spec file — the single source of truth. The offline
+self-test enforces that every policy row has a scenario and that every
+fixture number matches the REAL `GoalProgressEvaluator` + `GoalTrackPolicy`
+(`lib/features/goals/evaluation/`), so the fixtures double as the executable
+spec of the deterministic tier.
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply docs/evaluations/goal_agent_models/README.md:33]: Fixed in c318c232f — README now says P1–P15, and the offline self-test enforces that every policy row has a scenario, so the number cannot silently drift again.

File: docs/implementation_plans/2026-08-08_goal_agents_design.md [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+| `specImpliesCadenceAndDrifted` | `upsert_standing_agreement` (ChangeSet-gated) |
+| always | `record_goal_observation`, `remember`, `search_memory`, `set_next_check_in` (bounded/day) |
+
+The FACTS block rendered into the prompt is exactly the shape the eval
+fixtures author (`buildStepsFacts`/`buildGymFacts`) — the eval is the
+contract; the runtime renders the same JSON from real registers.
+
+## 2. Ad pipeline
+
+```
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply docs/implementation_plans/2026-08-08_goal_agents_design.md]: Fixed in c318c232f — fence marked `text`.

File: docs/implementation_plans/2026-08-08_goal_agents_kickoff_plan.md [hunk 6]
```diff
@@ -0,0 +1,677 @@
...
+insufficientData→report, no ad, never fabricate gap values; habit completed while "behind" ad
+live→retire, no new ad same wake; unanswered user message→dialogue first; clear change
+request→propose exactly once, restate current goal first; vague musing→clarifying question, no
+proposal.
+
+**16 scenarios** (gp_on_track/slightly_off/badly_off/recovering/data_gap/noop/gh_gym_pace;
+ad_create_off_track/leakage_pressure/stale_after_completion/no_double; evo_replace_metric/
+adjust_target/ambiguous/withdrawn-multi-turn; wk_dialogue_over_report). New predicate fields
+N1–N4: `followUpUserMessages`, `requiredAssistantContentTermGroups`/`forbiddenAssistantContent
+Claims`, `maxToolCallCounts`, `numberTerms()` formatting-variant helper. Everything else = existing
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply docs/implementation_plans/2026-08-08_goal_agents_kickoff_plan.md]: Fixed in c318c232f — the digest is kept verbatim as the received session record, now prefixed with a supersession note pointing at the shipped contract (6 tools + cta, P1–P15, 23 scenarios) and naming goal_agent_spec.dart as authoritative.

File: docs/implementation_plans/2026-08-08_goal_agents_kickoff_plan.md [hunk 7]
```diff
@@ -0,0 +1,677 @@
...
+histories; search; a11y; tokens; l10n; sync constraints; migration plan). Not built this session.
+
+### Step 7 — Verification pass & wrap-up
+Run the checks below, then present the document set + eval run book for review. No commits unless
+requested.
+
+## Verification
+
+- `dart-mcp.analyze_files` → **zero warnings/infos** (repo policy); `fvm dart format .`
+  (not dart-mcp format). Register repo root via `dart-mcp.add_roots` first.
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply docs/implementation_plans/2026-08-08_goal_agents_kickoff_plan.md:661]: Declining this one: the repo's AGENTS.md explicitly mandates `fvm dart format .` and states "Do not use dart-mcp.dart_format" — the plan text follows the repo guideline as written.

File: docs/implementation_plans/2026-08-08_reusable_chat_interface_requirements.md [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+  behind `enableAiStreamingFlag` once traced end-to-end.
+- REQ-14 A failed turn must surface as a retryable error bubble carrying
+  the wake-run error class — never a silently dropped message.
+
+## 7. Pagination & scale
+
+- REQ-15 Reverse list, createdAt-cursor pagination (~50/page), lazy
+  payload hydration. Must stay responsive on a **ten-year history**
+  (tens of thousands of rows): no full-log reads on open (ADR 0057's
+  bounded-read invariant applies to the UI too).
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply docs/implementation_plans/2026-08-08_reusable_chat_interface_requirements.md]: Fixed in c318c232f — REQ-15 now specifies a lexicographic compound cursor over (createdAt, stableEntityId), matching REQ-22's tiebreak.

File: lib/features/goals/evaluation/goal_progress_evaluator.dart [hunk 9]
```diff
@@ -0,0 +1,441 @@
...
+        satisfied = satisfiedCount > 0;
+        target = 1;
+        pace = _paceAny(children);
+      case _CompositeKind.atLeastCount:
+        final sorted = [...ratios]..sort((a, b) => b.compareTo(a));
+        final top = sorted.take(math.max(1, successes)).toList();
+        ratio = top.reduce((a, b) => a + b) / top.length;
+        satisfied = satisfiedCount >= successes;
+        target = successes;
+        pace = _paceAll(children);
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/features/goals/evaluation/goal_progress_evaluator.dart]: Good catch — real correctness bug. Fixed in c318c232f: atLeastCount gets its own `_paceAtLeast` instead of borrowing allOf semantics. The quota is dead only when fewer than `successes` children remain *possible* (satisfied, or not ruled out by their own pace); affirmatively on pace when at least `successes` are satisfied or feasible. Tests cover 2-of-3 with one dead leg (now feasible), the same tree as allOf (still sinks), and 2-of-3 with two dead legs (correctly impossible).

File: lib/features/goals/model/goal_criterion.dart [hunk 10]
```diff
@@ -0,0 +1,233 @@
...
+    required String criterionId,
+    required List<GoalCriterion> criteria,
+    required int successes,
+    String? title,
+  }) = GoalCriterionAtLeastCount;
+
+  const GoalCriterion._();
+
+  factory GoalCriterion.fromJson(Map<String, dynamic> json) =>
+      _$GoalCriterionFromJson(json);
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/features/goals/model/goal_criterion.dart:84]: Partially addressed, remainder scoped: the evaluator already throws deterministically on empty composites, and `periodRange` now rejects non-positive rolling counts (c318c232f). The full decode-time validator (fractional `targetCount`/`successes`, `successes > criteria.length`, malformed targets) belongs to the entity/persistence layer, which this PR deliberately does not contain — it is now an explicit deliverable of the design doc's PR-1 row rather than an implicit hope.

File: lib/features/goals/model/goal_window.dart [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+sealed class GoalWindow with _$GoalWindow {
+  /// A single calendar day.
+  const factory GoalWindow.day() = GoalWindowDay;
+
+  /// A trailing window of [count] days ending on the evaluation day.
+  ///
+  /// Each evaluation day anchors its own period ("rolling 7-day average"),
+  /// so [periodKey] is the day key of the reference date.
+  const factory GoalWindow.rollingDays({required int count}) =
+      GoalWindowRollingDays;
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/features/goals/model/goal_window.dart:23]: Fixed in c318c232f — `periodRange` throws ArgumentError for count < 1 (an inverted range must fail loudly, not evaluate to nonsense), tested both via the constructor and via `fromJson` with a malformed payload.

File: scripts/goal_agent_eval_matrix.sh [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+
+echo "Running $SAMPLES sample(s) per model, up to $MAX_PARALLEL at a time."
+start=$(date +%s)
+for model in $MODELS; do
+  for sample in $(seq 1 "$SAMPLES"); do
+    while (( $(jobs -rp | wc -l) >= MAX_PARALLEL )); do wait -n; done
+    run_one "$model" "$sample" &
+  done
+done
+wait
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply scripts/goal_agent_eval_matrix.sh]: Declining for now: this is the identical throttling pattern used by the existing scripts/penguin_wake_eval_matrix.sh, and eval dev machines run modern bash. If bash-3.2 portability is ever needed, both matrix scripts should change together rather than diverge.

File: test/features/agents/eval/goal/goal_agent_eval_live_test.dart [hunk 13]
```diff
@@ -0,0 +1,243 @@
...
+              jsonEncode({
+                'headline': args?['headline'],
+                'altText': args?['altText'],
+                'tone': args?['tone'],
+              }),
+            );
+            stdout.writeln('ad image: $path');
+          }
+        }
+      }
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/agents/eval/goal/goal_agent_eval_live_test.dart:196]: Fixed in c318c232f — each image generation is wrapped in try/catch, logs 'ad image FAILED for <scenario> × <model>: <error>' and continues; one Gemini hiccup can no longer abort the optional stage after the scoring artifacts exist.

File: test/features/agents/eval/goal/goal_agent_eval_runner_test.dart [hunk 14]
```diff
@@ -0,0 +1,344 @@
...
+          call(
+            GoalAgentToolNames.updateGoalReport,
+            '{"status":"insufficientData","oneLiner":"You slacked off.", '
+            '"tldr":"Looks like you fell behind badly this week."}',
+          ),
+        ],
+        assistantContent: '',
+      );
+      expect(category, GoalAgentEvalFailureCategory.forbiddenReportContent);
+    });
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/agents/eval/goal/goal_agent_eval_runner_test.dart]: Fixed in c318c232f — renamed to 'guilt-tripping claims in a data-gap report are caught', which is what the assertions actually exercise.

File: test/features/agents/eval/goal/support/goal_agent_spec.dart [hunk 15]
```diff
@@ -0,0 +1,362 @@
...
+/// Track-status vocabulary of the report tool — the deterministic
+/// [GoalTrackStatus] enum verbatim, so the evaluator, the agent contract and
+/// the eval assertions cannot drift apart.
+final List<String> goalTrackStatusNames = [
+  for (final status in GoalTrackStatus.values) status.name,
+];
+
+/// The draft system prompt. Deliberately lean (~1.4k chars): the payload
+/// lesson from task-agent evals is that long prompts get skimmed, and every
+/// number the model needs arrives in the wake FACTS block, not here.
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/agents/eval/goal/support/goal_agent_spec.dart]: Fixed in c318c232f — the comment now states the real budget: hard-capped at 3.2k chars by the offline test.

File: test/features/ai/eval/support/eval_text_matchers.dart [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+    final stop = end + _claimNegationWindow >= normalizedText.length
+        ? normalizedText.length
+        : end + _claimNegationWindow;
+    // Skip the claim itself so a cue inside it cannot excuse the claim.
+    final context =
+        '${normalizedText.substring(start, index)} '
+        '${normalizedText.substring(end, stop)}';
+    if (!_claimNegationPattern.hasMatch(context)) return true;
+    index = normalizedText.indexOf(needle, end);
+  }
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/ai/eval/support/eval_text_matchers.dart:100]: Nice edge case — fixed in c318c232f: partial words at either window edge are dropped before cue matching (a severed 'casino' can no longer donate a whole-word 'no'). New eval_text_matchers_test.dart includes exactly that construction plus a control proving intact cues still negate.

File: tool/goal_agent_eval_report.dart [hunk 17]
```diff
@@ -0,0 +1,158 @@
...
+      'Usage: dart run tool/goal_agent_eval_report.dart <artifact.json>...',
+    );
+    exitCode = 2;
+    return;
+  }
+  final artifacts = [
+    for (final path in paths)
+      jsonDecode(File(path).readAsStringSync()) as Map<String, dynamic>,
+  ];
+  stdout.write(buildGoalAgentEvalMergedReport(artifacts));
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply tool/goal_agent_eval_report.dart:39]: Fixed in c318c232f — the CLI now exits 2 with a clear 'did every eval process fail?' message when an argument doesn't exist (the unexpanded-glob case), and the matrix script uses nullglob with an explicit empty-set guard before invoking it.

File: scripts/goal_agent_eval_matrix.sh [hunk 18]
```diff
@@ -0,0 +1,78 @@
...
+  echo "Set MELIOUS_API_KEY (or GOAL_AGENT_EVAL_API_KEY)." >&2
+  exit 2
+fi
+
+mkdir -p "$LOG_DIR"
+
+# Warm the build once. Otherwise every process in the first wave compiles the
+# same test concurrently, which is slower than compiling it once.
+echo "Warming the test build..."
+LOTTI_GOAL_AGENT_EVAL_LIVE= fvm flutter test "$TEST_PATH" >/dev/null 2>&1
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply scripts/goal_agent_eval_matrix.sh]: Fixed in 06d037f8d — explicit `LOTTI_GOAL_AGENT_EVAL_LIVE=''`.

File: scripts/goal_agent_eval_matrix.sh [hunk 19]
```diff
@@ -0,0 +1,78 @@
...
+#
+# Override the model list with GOAL_AGENT_EVAL_MODELS (space separated).
+# Cost figures in the report are observations for monitoring — never caps.
+set -uo pipefail
+
+SAMPLES="${1:-3}"
+MAX_PARALLEL="${2:-4}"
+TEST_PATH="test/features/agents/eval/goal/goal_agent_eval_live_test.dart"
+OUT_DIR="eval_artifacts"
+LOG_DIR="$OUT_DIR/logs"
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply scripts/goal_agent_eval_matrix.sh]: Good catch — real report-corruption bug. Fixed in 06d037f8d: each invocation writes into `eval_artifacts/goal_agent_<timestamp>/` and the merge globs only that run's directory, so stale artifacts from earlier runs can never mix into the totals.

File: scripts/goal_agent_eval_matrix.sh [hunk 20]
```diff
@@ -0,0 +1,78 @@
...
+#
+# Reads credentials from the environment:
+#   MELIOUS_API_KEY (or GOAL_AGENT_EVAL_API_KEY)
+#
+# Override the model list with GOAL_AGENT_EVAL_MODELS (space separated).
+# Cost figures in the report are observations for monitoring — never caps.
+set -uo pipefail
+
+SAMPLES="${1:-3}"
+MAX_PARALLEL="${2:-4}"
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply scripts/goal_agent_eval_matrix.sh:21]: Fixed in 06d037f8d — both arguments are validated as positive integers before any job launches.

File: scripts/goal_agent_eval_matrix.sh [hunk 21]
```diff
@@ -0,0 +1,78 @@
...
+#
+# Usage:
+#   scripts/goal_agent_eval_matrix.sh [samples] [max_parallel]
+#
+# Reads credentials from the environment:
+#   MELIOUS_API_KEY (or GOAL_AGENT_EVAL_API_KEY)
+#
+# Override the model list with GOAL_AGENT_EVAL_MODELS (space separated).
+# Cost figures in the report are observations for monitoring — never caps.
+set -uo pipefail
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply scripts/goal_agent_eval_matrix.sh:18]: Fixed in 06d037f8d — `run_one` returns non-zero on failure, every `wait -n` collects statuses into a failure counter, the final drain loop does the same, a failed merge exits 1, and the script exits non-zero when any job failed (after still writing the merged report for the jobs that succeeded).

File: docs/implementation_plans/2026-08-08_goal_agents_design.md [hunk 22]
```diff
@@ -0,0 +1,281 @@
...
+
+| Fact | Tool exposed |
+| --- | --- |
+| `statusTransitioned` | `update_goal_report` (absent tool = wake CANNOT churn) |
+| `hasActiveNudge` | `retire_goal_ad` |
+| `offTrackOrWorseningAndNoFreshNudge` | `create_goal_ad` |
+| ...and `topRatedReusableNudges` non-empty | `rerun_goal_ad` (offered alongside create; prompt prefers it) |
+| `inDialogue` / clear change request | `propose_goal_revision` (ChangeSet-gated) |
+| `specImpliesCadenceAndDrifted` | `upsert_standing_agreement` (ChangeSet-gated) |
+| always | `record_goal_observation`, `remember`, `search_memory`, `set_next_check_in` (bounded/day) |
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply docs/implementation_plans/2026-08-08_goal_agents_design.md]: Fixed in 06d037f8d — the design doc now states explicitly that goal_agent_spec.dart is the SINGLE Phase B tool contract: the gate rows for upsert_standing_agreement, remember/search_memory and set_next_check_in are marked as PR-4/6/2 additions outside the six-tool eval contract, with the rule that any later tool lands in the spec + scenarios first. The kickoff gate digest gains rerun_goal_ad and the implementation checklist is labeled historical with a pointer to the authoritative spec.

File: docs/implementation_plans/2026-08-08_goal_agents_kickoff_plan.md [hunk 23]
```diff
@@ -0,0 +1,680 @@
...
+  in one tap). The evaluator is the codebase's FIRST rule-tree evaluator; AutoCompleteRule adapter
+  becomes trivial later.
+- `goalProgress` keyed register (id `goal_progress:<agentId>:<periodKey>`, recomputed-never-
+  accumulated like weekRollup → LWW-convergent across devices; trackStatus onTrack|atRisk|offTrack|
+  achieved|insufficientData as subtype for indexed scans; attainment 0..1 + per-criterion results;
+  ~1MB/goal-decade, retention-exempt — it IS the chartable history + cheap agent context).
+- `goalNudge` ad entity (status draft→ready→active→dismissed|expired|superseded|failed as subtype;
+  `GoalNudgeBrief` typed brief = the ONLY payload the image request may see; imageEntryId
+  (JournalImage → media sync free); headline/caption composited ON DEVICE, never sent;
+  triggerProgressId evidence; staleAt pulled forward by habit completions; reasonSummary).
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply docs/implementation_plans/2026-08-08_goal_agents_kickoff_plan.md:427]: Fixed in 06d037f8d — the kickoff digest's register list now carries the full GoalTrackStatus vocabulary (recovering included) and the nudge lifecycle includes retired (the reuse re-entry from ADR 0055 Decision 7); both annotated rather than silently rewritten, since the section is a session record.
